solutions/devsprint-viewcode-1/ViewCodeChallenge-RealEstate/Screens/Components/CarouselView/Cells/CarouselCollectionViewCell.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  CarouselCollectionViewCell.swift
+//  ViewCodeChallenge-RealEstate
+//
+//  Created by Elena Diniz on 19/10/21.
+//
+
+import UIKit
+
+class CarouselCollectionViewCell: UICollectionViewCell {
+    
+    static let cellIdentifier = "CarouselViewCell"
+    
+    lazy var imageView: UIImageView = {
+       
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.contentMode = .scaleAspectFill
+        return imageView
+    }()
+    
+    override init(frame: CGRect = .zero) {
+        super.init(frame: frame)
+        configureSubviews()
+        configureConstraints()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+//MARK: - Setup View Cell
+extension CarouselCollectionViewCell: ViewProtocol {
+    func configureSubviews() {
+        self.contentView.addSubview(self.imageView)
+    }
+    
+    func configureConstraints() {
+        NSLayoutConstraint.activate([
+            self.imageView.topAnchor.constraint(equalTo: self.contentView.topAnchor),
```

Comment:
Essas constraints provavelmente vão entrar em conflito. Se por exemplo o frame que esta célula estiver contida tiver um `width` for maior que 400 a constraint `widthAnchor` vai entrar em conflito com a `trailingAnchor `.

A mesma coisa pode acontecer com as constraints `heightAnchor` e `bottomAnchor`.

Reply:
Done ✅